core/trino-parser/src/main/java/io/trino/sql/ExpressionFormatter.java

```diff
@@ -727,17 +726,12 @@ private String visitFilter(Expression node, Void context)
         @Override
         protected String visitQuantifiedComparisonExpression(QuantifiedComparisonExpression node, Void context)
         {
-            return new StringBuilder()
-                    .append("(")
-                    .append(process(node.getValue(), context))
-                    .append(' ')
-                    .append(node.getOperator().getValue())
-                    .append(' ')
-                    .append(node.getQuantifier().toString())
-                    .append(' ')
-                    .append(process(node.getSubquery(), context))
-                    .append(")")
-                    .toString();
+            return format(
+                    "(%s %s %s %s)",
+                    process(node.getValue(), context),
+                    node.getOperator().getValue(),
+                    node.getQuantifier(),
+                    process(node.getSubquery(), context));
```

Comment:
`"...".formatted(...)`

Reply:
`String#formatted` was introduced in Java 13, but `trino-parser` is deliberately at Java 8 language level.

Reply:
After `trino-grammar` was extracted from `trino-parser`, this is now possible.